PuzzleSlider/src/Engine/GameData.java

```diff
@@ -124,6 +137,23 @@ public GridPoint getEmptyTile() {
 		return mBlankSpot;
 	}
 	
+	public int getMoveCount() {
+		return mMoveCount;
+	}
+	
+	public int incrementMoveCount(){
+		mMoveCount++;
+		return mMoveCount;
```

Comment:
The idea was maybe in future it would be handy to immediately return incremented value and not use incerement and getter functions.